compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeNarrower.java

```diff
@@ -380,7 +397,7 @@ private NarrowedTypes getNarrowedTypesForBinaryOp(Map<BVarSymbol, NarrowedTypes>
             falseType = getTypeUnion(lhsFalseType, tmpType);
         } else {
             BType tmpType = types.getTypeIntersection(nonLoggingContext, lhsFalseType, rhsTrueType, this.env);
-            trueType = getTypeUnion(lhsTrueType, tmpType);
+            trueType = isLhsOperand ? getTypeUnion(lhsTrueType, tmpType) : getTypeUnion(tmpType, lhsTrueType);
```

Comment:
Not related to this PR, but `getTypeUnion` or at least the argument names are a bit confusing. With a name like `getTypeUnion` IMO, one wouldn't think the argument order matters. Shall we create an issue for this also?

Reply:
Created https://github.com/ballerina-platform/ballerina-lang/issues/33569